FirebaseDatabaseSwift/Tests/Codable/RTDBEncoderTests.swift

```diff
@@ -0,0 +1,509 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import FirebaseDatabase
+import FirebaseDatabaseSwift
+import XCTest
+
+class FirebaseDatabaseDefaultingTests: XCTestCase {
+  override func setUp() {
+    Database.Encoder.defaultEncoder = {
+      let encoder = Database.Encoder()
+      encoder.keyEncodingStrategy = .convertToSnakeCase
+      return encoder
+    }
+
+    Database.Decoder.defaultDecoder = {
+      let decoder = Database.Decoder()
+      decoder.keyDecodingStrategy = .convertFromSnakeCase
+      return decoder
+    }
```

Comment:
As mentioned above with the default encoders, I think instantiating a new encoder for each test would be more clear. Then we can also test custom snakeCase encoding there as well.